### DISCUSSION TITLE
Cancel button at the Confirmation stage

### DISCUSSION THREAD

**Discussion by PVH-tc** [grahampugh/erase-install#481] [General]
I have a Jamf policy that triggers on Startup, Login, and Recurring Check-in. The Retry Event number is set to 3 attempts "On next selected triggered occurrence".

I have a deferral duration of 5 days.

I have a message set in the User Interaction's Complete Message.

When the popup for confirmation appears, there are the **Cancel** and **Confirm** buttons. If Cancel is clicked on, the Complete Message window appears but the policy's log still shows as Pending.
<img width="592" alt="Screenshot 2023-12-07 at 11 42 43 AM" src="https://github.com/grahampugh/erase-install/assets/111586928/36b4ebb8-82d9-4e4b-a69d-f732cb0681e7">

We want the confirmation window to appear. We are using this as their last chance to save their work before the update process prevents them from using their computer during the update.

I am concerned about a couple of behaviors.
1. The next selected triggered occurrence does not retry the policy.
2. The deferral deadline does not automatically update the scoped computer.
3. The policy log does not register or convert from Pending to Retrying.

Number 1 and 2 is more of me asking for assistance with troubleshooting. Number 3 could end up being an idea for the next version.

**Comment by liubkkkko** [comment]
I have a similar problem, I use the --confirm flag when updating my Macs to allow the user to run the update at their convenience. Sometimes my users click "cancel" on the pre-installation confirmation window after successfully downloading it. After that, the policy is marked as Completed and in the logs I see that the user canceled the installation. Does the program exit completely? Is it possible to restart the program only by repeating the policy, for example, once a week?

**Reply by grahampugh** [reply]
Hi there

If you want the user to run the policy at their convenience, why not put it in Self Service rather than use the --confirm option?

**Reply by liubkkkko** [reply]
We as administrators are more interested in updating the system than users, I want the user to be able to choose a convenient time for the update and somehow understand its inevitability. I think self-service accommodation will have less effect. Anyway, thanks for your work, your product is great

**Comment by PVH-tc** [comment]
I understand the --confirm switch prompts the user to continue or proceed. If they choose to **Cancel**, the Jamf policy logs the computer as **Completed** which is incorrect because the computer never updates.

After posting my original ask, I've removed the --confirm switch to prevent a cancellation. Ideally, we wanted the user to confirm but I need to get users' computer to move to the next stage of the update process.

If the user has an Apple Silicon computer, the popup for the username/password appears. There is a **Cancel** button there too. If they choose to **Cancel**, the Jamf policy logs the computer as **Completed** which is incorrect because the computer never updates.

Below is an example of the Jamf log for an Apple Silicon marked as Completed.
2024-03-28 08:07:31 | v32.0 | [erase-install] Running on architecture arm64
2024-03-28 08:07:31 | v32.0 | [get_user_details] ask for user credentials (attempt 1/5)
2024-03-28 08:07:32 | v32.0 | [get_default_dialog_args] Invoking utility dialog
yes Mxxx.xxxk is a member of everyone
2024-03-28 08:07:49 | v32.0 | [get_user_details] Mxxx.xxxk is a Volume Owner
<dscl_cmd> DS Error: -14090 (eDSAuthFailed)
2024-03-28 08:07:49 | v32.0 | [check_password] ERROR: The password entered is NOT the login password for Mxxx.xxxk.
2024-03-28 08:07:50 | v32.0 | [get_user_details] ask for user credentials (attempt 2/5)
2024-03-28 08:07:50 | v32.0 | [get_default_dialog_args] Invoking utility dialog
2024-03-28 08:07:56 | v32.0 | [get_user_details] user cancelled dialog so exiting...

2024-03-28 08:07:56 | v32.0 | [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success

2024-03-28 08:07:56 | v32.0 | [finish] Script exit code: 0

To me, this seems to be a logic bug.

**Reply by grahampugh** [reply]
See https://github.com/grahampugh/erase-install/wiki/4.-Upgrading-or-reinstalling-macOS-without-wiping-the-system#--max-password-attempts

**Reply by PVH-tc** [reply]
Thank you Graham. I overlooked that setting.
